### PR TITLE
Fix Layout-Headers for Publish Try 2

### PR DESCRIPTION
**Pull request by NickGerleman**
Continue from #3887

Add a missing directory separator that wasn't in the env variable. ~Add
LayoutHeaders to a build flavor in the PR matrix so we can find this
earlier.~ (see update below)

###### Microsoft Reviewers: [Open in CodeFlow](http://wpcp.azurewebsites.net/CodeFlowProtocolProxy2.php?pullrequest=https://github.com/microsoft/react-native-windows/pull/3888)

### CODE REVIEW

File: vnext/Scripts/Tfs/Layout-Headers.ps1 [hunk 1]
```diff
@@ -12,14 +12,14 @@ param(
...
-    $intermediateBuildDir =  if ($env:BaseIntDir) { $env:BaseIntDir } else { "$SourceRoot\vnext\build\" }
+    $intermediateBuildDir =  if ($env:BaseIntDir) { $env:BaseIntDir } else { "$SourceRoot\vnext\build" }
     $relativeRnDir = @(gci $intermediateBuildDir react-native-patched -Recurse -Directory -Name)[0]
 
     if (!$relativeRnDir) {
         throw "Cannot find patched React Native Directory (has a project been built?)"
     }
 
-    $ReactNativeRoot = $intermediateBuildDir + $relativeRnDir
+    $ReactNativeRoot = $intermediateBuildDir + "\" + $relativeRnDir
```
Comment (JunielKatarn) [inline vnext/Scripts/Tfs/Layout-Headers.ps1]: Use `Join-Path` instead.

i.e.
```
Join-Path -Path $intermediateBuildDir -ChildPath $relativeRnDir
```

**Comment by NickGerleman** [comment]
@JunielKatarn there's some weirdness going on with CI. It looks like in prior builds most of "Build and Pack Nuget" completely no-oped, even for desktop. In this one we materialistically fail when trying to pack ReactUWP x64 debug, even though we ran a script to remove ReactUWP from the NuSpec.

Still trying to figure out how adding LayoutHeaders to the build matrix could change this. Us no-oping on previous PRs seems like a bug though. Know what's going on here?

**Comment by NickGerleman** [comment]
E.g. this for an older build: 
image

**Comment by NickGerleman** [comment]
After this change (even for Desktop): 
image

**Comment by NickGerleman** [comment]
Going to create an issue for the Nuget Issues and just leave header layout out of the PR Matrix for now.

**Comment by NickGerleman** [comment]
@JunielKatarn could you take a look at this again?

**Pull request merged**